stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -0,0 +1,272 @@
+        total_records = int(offset) + int(length)
+
+        if self.api_client.result_limit < total_records:
+            total_records = self.api_client.result_limit
+
+        if not metadata:
+            if total_records <= Connector.NOZOMI_MAX_PAGE_SIZE:
+                page_size = total_records
+            else:
+                page_size = Connector.NOZOMI_MAX_PAGE_SIZE
+
+        return result_count, total_records, page_number, page_size
+
+    def handle_api_response(self, response_wrapper):
+        """
+        Handling response codes
+        :param response_wrapper, object
+        :return: response_dict, dict
+                 return_obj, dict
+        """
+        response_dict = {}
+        return_obj = {}
+        response = response_wrapper.read().decode('utf-8')
+        if response.startswith('{') and response.endswith('}'):
+            response_dict = json.loads(response)
+
+        if response_wrapper.code == 200:
+            # Handling invalid query exception
+            if 'Query is not valid' in response_dict.get('error', ''):
+                return_obj = self.handle_api_exception(400, response_dict.get('error'))  # changing response code to 400
+            # Handling expired token
+        elif response_wrapper.code == 401 and not self.expired_token_count and \
+                (response == '' or 'Signature has expired' in response_dict.get('error', {}).get('message', '')):
+            self.expired_token_count += 1
+        else:
+            error = response_dict.get('error', '')
+            if response_wrapper.code == 403:
+                error = 'Query length is too long or Invalid Query.' + error
+            return_obj = self.handle_api_exception(response_wrapper.code, error)
+        return response_dict, return_obj
+
+    @staticmethod
+    def get_page_size(result_count, total_records, page_size):
+        """
+        Finding page size based on remaining records
+        :param result_count, int
+        :param total_records, int
+        :param page_size, int
+        :return: remaining_records, int
+                 page_size, int
+        """
+        if result_count < total_records:
+            remaining_records = total_records - result_count
+        else:
+            return 0, page_size
+
+        if remaining_records > Connector.NOZOMI_MAX_PAGE_SIZE:
+            page_size = Connector.NOZOMI_MAX_PAGE_SIZE
+        else:
+            page_size = remaining_records
+
+        return remaining_records, page_size
+
+    @staticmethod
+    def handle_data(data, metadata, offset, total_records, return_obj):
+        """
+         Process the data
+        :param data, list
+        :param metadata, dict
+        :param offset, int
+        :param total_records, int
+        :param return_obj, dict
+        :return: return_obj, dict
+        """
+        if data:
+            data = Connector.get_results_data(data)
+            if metadata:
+                return_obj['data'] = data if data else []
```

Comment:
Page_size is static, so this can overflow above total_records length.

You can just do 
data[0: total_records]

Reply:
From CLI we can give custom offset values. If user gives offest=10 and length =1000, then we need to fetch 10th to 1010th record. We can't use offset always 0.

Reply:
Supporting offsets with metadata would be quite odd and I can't really think of a use case where this would be necessary. 

As a note, this comment was originally made with the idea of a page_size that does not change. Hence the need to account for a data list that could be greater than the requested amount. For example, If your page_size is 99 and the length is 100, then you could get a resulting data of size 198 (98 more than requested). You'd need to trim it down to 100 (or the total_records, which should usually be 0+length when using metadata).

Reply:
We are not using offset with metadata. 
Offset slicing is only used in first batch iteration not for next batch iterations.